### PR TITLE
NF: uses empty when possible

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
For the sake of readability

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 1]
```diff
@@ -1537,7 +1537,7 @@ private boolean onImagePaste(EditText editText, Uri uri) {
     private void setMMButtonListener(ImageButton mediaButton, final int index) {
         mediaButton.setOnClickListener(v -> {
             Timber.i("NoteEditor:: Multimedia button pressed for field %d", index);
-            if (mEditorNote.items()[index][1].length() > 0) {
+            if (!mEditorNote.items()[index][1].isEmpty()) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:1540]: This variable should be extracted, IMO it's harder to read after the change.

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java [hunk 2]
```diff
@@ -100,7 +100,7 @@ protected void onCreate(Bundle savedInstanceState) {
         initAllContentViews();
 
         SharedPreferences preferences = AnkiDroidApp.getSharedPrefs(getBaseContext());
-        if (preferences.getString("hkey", "").length() > 0) {
+        if (!preferences.getString("hkey", "").isEmpty()) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.java:103]: !TextUtils.isEmpty - null should also be handled here

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.java [hunk 3]
```diff
@@ -253,7 +253,7 @@ protected DecksFilter() {
             protected FilterResults performFiltering(CharSequence constraint) {
                 mFilteredDecks.clear();
                 ArrayList<SelectableDeck> allDecks = DecksArrayAdapter.this.mAllDecksList;
-                if (constraint.length() == 0) {
+                if (constraint.length() > 0) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.java:256]: ⚠️

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.java:256]: Reminder to self: Add a test here

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/TagsDialog.java [hunk 4]
```diff
@@ -341,7 +341,7 @@ private TagsFilter() {
             @Override
             protected FilterResults performFiltering(CharSequence constraint) {
                 mFilteredTags.clear();
-                if (constraint.length() == 0) {
+                if (constraint.length() > 0) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/TagsDialog.java:344]: Please stop mixing functional changes with `NF:` commits (presumably this was an intentional change as it's happened twice in the same code area in the same PR). 

This introduces a bug

File: AnkiDroid/src/main/java/com/ichi2/utils/JSONArray.java [hunk 5]
```diff
@@ -384,4 +384,8 @@ public Long next() {
             }
         };
     }
+
+    public boolean isEmpty() {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/utils/JSONArray.java:388]: `isNotEmpty()/any()` would also improve understanding

File: AnkiDroid/src/main/java/com/wildplot/android/parsing/AtomTypes/FunctionXAtom.java [hunk 6]
```diff
@@ -44,7 +44,7 @@ private boolean init(String funcString) {
 
             Pattern p = Pattern.compile("[^a-zA-Z0-9]");
             boolean hasSpecialChar = p.matcher(funcName).find();
-            if (!hasSpecialChar && (funcName.length() > 0)) {
+            if (!hasSpecialChar && (funcName.isEmpty())) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/wildplot/android/parsing/AtomTypes/FunctionXAtom.java:47]:
```suggestion
            if (!hasSpecialChar && (!funcName.isEmpty())) {
```

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
I don't think these mass readability improvements are worth the effort unless they're automated (ideally also linted so they don't happen again):

* It took time to write
* It took time to review (and will take time for re-review)
* It introduced bugs in code that was both working, and will hopefully have no reason to be read/changed (I've never touched the wildplot code).
* The benefit from the change is both for a small audience, and low impact

As a cost-benefit, it didn't seem worth it. 

But, let's get this in now it's been submitted.

**Comment by mikehardy** [comment]
No, let's please stop doing these for all the reasons stated. I have no time for this, sorry

**Pull request closed without merging**